### PR TITLE
feat: allow passing nativewindui components to cli as selectedComponents flag

### PR DESCRIPTION
**Pull request by dannyhw**
## Description

allows the user to pass a comma separated list of nativewind ui components like 
`--selected-components=date-picker,selectable-text` 

## Motivation and Context

this will also enable us to respect the selected components on the website version

## How Has This Been Tested?

ran it locally

## Screenshots (if appropriate):

image

### CODE REVIEW

File: cli/src/commands/create-expo-stack.ts [hunk 1]
```diff
@@ -316,15 +309,23 @@ const command: GluegunCommand = {
         const generateRerunScript = (cliResults: CliResults) => {
           let script = `npx create-expo-stack ${cliResults.projectName} `;
 
-          if (cliResults.packages.length > 0 && cliResults.packages[0].name === 'nativewindui') {
+          const nativewindUISelected = cliResults.packages.find((p) => p.name === 'nativewindui');
```
Comment (danstepanov) [inline cli/src/commands/create-expo-stack.ts:312]: This doesn't seem like a very good naming convention given how this value is now used to both assert that the user has selected NativeWindUI as a styling option AND check which components they have selected.

What are your thoughts on the following approach instead?
```
// Check if NativeWindUI is selected as the styling package
const isNativeWindUISelected = cliResults.packages.some((p) => p.name === 'nativewindui');

// Get the selected NativeWindUI package details (including components)
const nativeWindUIComponents = cliResults.packages.find((p) => p.name === 'nativewindui')?.options?.selectedComponents || [];
```

File: cli/src/commands/create-expo-stack.ts [hunk 2]
```diff
@@ -316,15 +309,23 @@ const command: GluegunCommand = {
...
+
+          const navigationSelected = cliResults.packages.find((p) => p.type === 'navigation');
+
+          if (cliResults.packages.length > 0 && nativewindUISelected) {
```
Comment (danstepanov) [inline cli/src/commands/create-expo-stack.ts:316]:
```suggestion
          if (isNativeWindUISelected) {
```

File: cli/src/commands/create-expo-stack.ts [hunk 3]
```diff
@@ -316,15 +309,23 @@ const command: GluegunCommand = {
...
             script += '--nativewindui ';
-            if (cliResults.packages[0].options.selectedComponents.length === 0) {
+
+            if (nativewindUISelected.options.selectedComponents.length === 0) {
```
Comment (danstepanov) [inline cli/src/commands/create-expo-stack.ts:319]:
```suggestion
            if (nativeWindUIComponents.length === 0) {
```

File: cli/src/commands/create-expo-stack.ts [hunk 4]
```diff
@@ -316,15 +309,23 @@ const command: GluegunCommand = {
...
               script += '--blank ';
+            } else if (nativewindUISelected.options.selectedComponents.length !== nativewindUIOptions.length) {
```
Comment (danstepanov) [inline cli/src/commands/create-expo-stack.ts:321]: just to confirm, in the event that the user selects all of the NativeWindUI components, they would just see a re-run script of `npx create-expo-stack@latest --nativewindui`, correct?

Versus if they select at least 1 component but less than all of the components, in which case they'll see `npx create-expo-stack@latest --nativewindui --selected-components={...}`?

File: cli/src/utilities/runCLI.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { Toolbox } from 'gluegun/build/types/domain/toolbox';
 import { confirm, isCancel, cancel, multiselect, select, text } from '@clack/prompts';
 
-import { defaultOptions } from '../constants';
+import { defaultOptions, nativewindUIOptions } from '../constants';
```
Comment (danstepanov) [inline cli/src/utilities/runCLI.ts:4]:
```suggestion
import { defaultOptions, nativeWindUIOptions } from '../constants';
```

File: cli/src/utilities/runCLI.ts [hunk 6]
```diff
@@ -298,24 +297,10 @@ export async function runCLI(toolbox: Toolbox, projectName: string): Promise<Cli
...
-        'picker',
-        'progress-indicator',
-        'ratings-indicator',
-        'selectable-text',
-        'slider',
-        'text',
-        'toggle'
-      ]
-    })) as SelectedComponents[];
+      initialValues: nativewindUIOptions
```
Comment (danstepanov) [inline cli/src/utilities/runCLI.ts:302]:
```suggestion
      initialValues: nativeWindUIOptions
```

File: cli/src/commands/create-expo-stack.ts [hunk 7]
```diff
@@ -11,7 +11,7 @@ import {
   runIgnite,
   showHelp
 } from '../utilities';
-import { DEFAULT_APP_NAME, defaultOptions } from '../constants';
+import { DEFAULT_APP_NAME, defaultOptions, nativewindUIOptions } from '../constants';
```
Comment (danstepanov) [inline cli/src/commands/create-expo-stack.ts:14]:
```suggestion
import { DEFAULT_APP_NAME, defaultOptions, nativeWindUIOptions } from '../constants';
```

File: cli/src/commands/create-expo-stack.ts [hunk 8]
```diff
@@ -203,29 +203,22 @@ const command: GluegunCommand = {
         } else if (options.nativewindui) {
           cliResults = clearStylingPackages(cliResults);
           cliResults = clearNavigationPackages(cliResults);
+
+          const parsedComponents = options.selectedComponents
+            .split(',')
+            .map((c: string) => c.trim())
+            .filter((item) => nativewindUIOptions.includes(item));
```
Comment (danstepanov) [inline cli/src/commands/create-expo-stack.ts:210]:
```suggestion
            .filter((item) => nativeWindUIOptions.includes(item));
```

File: cli/src/commands/create-expo-stack.ts [hunk 9]
```diff
@@ -203,29 +203,22 @@ const command: GluegunCommand = {
...
+
+          const selectedComponents = parsedComponents.length ? parsedComponents : nativewindUIOptions;
```
Comment (danstepanov) [inline cli/src/commands/create-expo-stack.ts:212]:
```suggestion
          const selectedComponents = parsedComponents.length ? parsedComponents : nativeWindUIOptions;
```

**Review by danstepanov — CHANGES_REQUESTED** [review]
couple nits

File: cli/src/commands/create-expo-stack.ts (hunk 1 above)
Comment (dannyhw) [reply cli/src/commands/create-expo-stack.ts]: sounds better to me 👍

File: cli/src/commands/create-expo-stack.ts (hunk 4 above)
Comment (dannyhw) [reply cli/src/commands/create-expo-stack.ts]: Yes what I aimed for. If you have everything selected it uses the default selection (everything). 

If you have unselected anything i.e more than 1 but less than the total number of options then we give them the list of components that were selected.

We could just always provide the list but it seems cleaner to not include the massive list when we don't need to

**Comment by dannyhw** [comment]
@danstepanov tested a few more scenarios and applied a few more fixes. Hopefully more readable now too. Thanks for the feedback 👍 .

**Comment by dannyhw** [comment]
thinking about adding nativewindui to the tests

**Pull request merged**